### ISSUE TITLE
Integer mask input to batched_mask_to_box() produces incorrect results

### ISSUE THREAD

**Issue by GenevieveBuckley** [facebookresearch/segment-anything#552]
The function `batched_mask_to_box()` from `segment_anything/utils/amg.py` produces incorrect results in cases where the mask input array has integer dtype (instead of dtype bool, which does work correctly)

## Minimum reproducible example
```python
import numpy as np
import torch
from segment_anything.utils.amg import batched_mask_to_box

mask_numpy = np.zeros((10,10)).astype(bool)
mask_numpy[7:9, 3:5] = True
expected_result_values = [3, 7, 4, 8]

device = "cpu"

# boolean mask input, works as expected
mask = torch.as_tensor(mask_numpy, dtype=torch.bool, device=device)
expected_result = torch.as_tensor(expected_result_values, dtype=torch.int, device=device)
result_from_boolean = batched_mask_to_box(mask)
assert all(result_from_boolean == expected_result)  # works as expected
print(result_from_boolean)
# tensor([3, 7, 4, 8])

# integer mask input, produces completely different box result
mask = torch.as_tensor(mask_numpy, dtype=torch.int, device=device)
expected_result = torch.as_tensor(expected_result_values, dtype=torch.int, device=device)
result_from_integer = batched_mask_to_box(mask)
assert all(result_from_integer == expected_result)  # fails with AssertionError
# AssertionError:
print(result_from_integer)
# tensor([-17, -13,   4,   8])
```

The function `batched_mask_to_box()` from segment-anything:

https://github.com/facebookresearch/segment-anything/blob/6fdee8f2727f4506cfbbe553e23b895e27956588/segment_anything/utils/amg.py#L303-L346
